### PR TITLE
Pipeline Create, Read, Update, Delete

### PR DESCRIPTION
**Pull request by mtarng**
Needs testing, but CRUD seems to be working for Pipelines.

Thanks for the help and support @madelineliao @tasdevani21 @nmiodice

### CODE REVIEW

**Comment by mtarng** [comment]
@nmiodice any chance we can get this merged?

**Comment by nmiodice** [comment]
@mtarng I rebased off of the current master which had changed. Waiting on a review from someone else from the team (I added reviewers to the PR)

File: terraform-provider/provider-src/src/provider.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ func Provider() *schema.Provider {
 			// Why is the key/value named the way they are?
 			"azuredevops_foo":      resourceFoo(),
 			"azuredevops_pipeline": resourcePipeline(),
-			"azuredevops_project": resourceProject(),
+			"azuredevops_project":  resourceProject(),
```
Comment (rguthriemsft) [inline terraform-provider/provider-src/src/provider.go]: Looks like an extra space got added

Comment (nmiodice) [reply terraform-provider/provider-src/src/provider.go]: This was the result of running a `go fmt ./...`.

**Pull request merged**